LunaDll/Misc/RuntimeHookComponents/RuntimeHookHooks.cpp

```diff
@@ -1602,6 +1602,27 @@ Characters playerStoredCharacters[] = {CHARACTER_MARIO,CHARACTER_MARIO,CHARACTER
 
 void __stdcall runtimeHookLoadLevel(VB6StrPtr* filename)
 {
+    std::wstring filenameWS = static_cast<std::wstring>(*filename);
+    size_t findLastSlash = filenameWS.find_last_of(L"/\\");
+    std::wstring dirWS = filenameWS.substr(0U, findLastSlash);
+    std::wstring nameWS = filenameWS.substr(findLastSlash + 1);
+    std::string dirST = WStr2Str(dirWS);
+    std::string nameST = WStr2Str(nameWS);
+
+    //If the legacy title screen is about to boot, prevent that and go straight to loading the episode
+    if(nameST == "intro.lvl" && dirST == gAppPathUTF8)
```

Comment:
I'm not too familiar with the underlying code of SMBX but shouldn't there be a flag, level mode or specific point of code execution that indicates if we're loading the title screen? Something more robust than going by filename. How does this if check handle exiting an episode via the pause menu or Misc.exitGame()?

Reply:
I don't think there is, unless I'm wrong. This is the earliest we can detect loading the title screen without directly loading intro.lvl.

Also the directory check is there to prevent any other lvl files from other places (Like a world folder) having the same name, be prevented to be loaded.

Reply:
there's `GM_LEVEL_MODE` which is equal to `-1` during the title screen.